craft_parts/packages/normalize.py

```diff
@@ -157,35 +157,59 @@ def _try_copy_local(path: Path, target: Path) -> bool:
 
 
 def fix_pkg_config(
-    root: Path,
-    pkg_config_file: Path,
-    prefix_trim: Optional[Path] = None,
+    prefix_prepend: Path, pkg_config_file: Path, prefix_trim: Optional[Path] = None
 ) -> None:
-    """Rewrite the prefix entry in a pkg-config file.
-
-    :param root: The root to add to the configuration prefix.
-    :param pkg_config_file: The pkg-config file to process.
-    :param prefix_trim: The initial path to remove from the configuration prefix.
+    """Fix the prefix parameter in pkg-config files.
+
+    This function does 3 things:
+    1. Remove `prefix_trim` from the prefix.
+    2. Remove directories commonly added by staged snaps from the prefix.
+    3. Prepend `prefix_prepend` to the prefix.
+
+    The prepended directory depends on the source of the pkg-config file:
+    - From snaps built via launchpad: the stage directory
+      `/build/<snap-name>/stage/` is prepended
+    - From snaps built via a provider: the stage directory `/root/stage/` is prepended
+    - Built during the build stage: the install directory may be prepended
+
+    :param pkg_config_file: pkg-config (.pc) file to modify
+    :param prefix_prepend: directory to prepend to the prefix
+    :param prefix_trim: directory to remove from prefix
     """
-    # FIXME: see https://bugs.launchpad.net/snapcraft/+bug/1916281
-    pattern_trim = None
+    # build patterns
+    prefixes_to_trim = [r"/build/[\w\-. ]+/stage", "/root/stage"]
```

Comment:
Can we avoid hard-coding patterns based on specific usage cases, and do something based on our pathname variables instead? E.g. providers use `/root/stage` but this value comes from `info.stage_dir`.

Reply:
After discussing, these were our conclusions:
1. `/root/stage` and `/build/<snap-name>/stage` are hard-coded into pkg-config files in snaps already in the snap store.
2. Because of this, there is not an ideal way to trim out these prefixes without hard-coding these patterns.
3. A future snapcraft improvement would be to trim out these prefixes during the final packing step (like in the `setup_assets()` function).